Upgrade to atom-linter v2

binore opened the pull request:
Fixed bug with relative path
Changed config and issue objects for v2
Removed coala.less (now using atom styles instead of our ones) 

Closes https://github.com/coala/coala-atom/issues/59

Commit:
Upgrade to atom-linter v2

Fixed bug with relative path
Changed config and issue objects for v2
Removed coala.less (now using atom styles instead of our ones) 

Closes https://github.com/coala/coala-atom/issues/59

Commit:
Fix main-spec.js

Udayan12167 commented:
I think your first commit should be split into three commits as it changes three different files?

ghost commented:
May be, i'll do that way next time

sils commented on lib/main.coffee:
```diff
@@ -43,12 +43,12 @@ module.exports =
         parameters = []
         parameters.push '--find-config'
-        parameters.push '--limit-files=' + filePath
+        parameters.push '--limit-files=' + path.basename(filePath)
```
why the basename? An absolute path will always work, the basename may well not be working?

binore replied:
but we execute with project path as workingDirectory for process, so we dont need absolute path

sils replied:
the basename is usually the part after the last slash right? So it's not necessarily relative to the project directory.

binore replied:
as i said, coala will be executed with workingDirectory = directory, which contains file for analysis. So, we need to pass only file name in this direcory

sils commented on lib/main.coffee:
```diff
@@ -43,12 +43,12 @@ module.exports =
       grammarScopes: ['*']
       scope: 'file'
-      lintOnFly: true
+      lintsOnChange: true
```
actually why was this on true? We only want on save right?

binore replied:
Pls, look at this https://github.com/coala/coala-atom/issues/62

sils replied:
ah because of the file opening, makes sense :+1:

sils commented on lib/main.coffee:
```diff
@@ -22,9 +22,9 @@ module.exports =
-      'Normal',  # Result severity 1 = Normal
-      'Major'    # Result severity 2 = Major
+      'info',    # Info
+      'warning',  # Normal
```
comment not aligned with the others

sils reviewed (changes requested):
FWIW I'd just squash them together,it's a good logical change.

Commit:
main.coffee: disable lints on change

Set lintsOnChange false cause we don't need to analyze file after every change

satwikkansal commented on lib/main.coffee:
```diff
@@ -22,9 +22,9 @@ module.exports =
+      'info',     # Info
       'warning',  # Normal
-      'error'    # Major
+      'error'     # Major
```
this indentation change not relevant to the commit.

satwikkansal commented:
Please describe what exactly are you fixing in the commit message of https://github.com/coala/coala-atom/pull/63/commits/4e4c3d229d674dfb9fecb2aecc67b57af0762946

sils commented:
I just squashed it with a new commit message because it's not too important here and it's way more important to get this merged
